crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -902,9 +931,10 @@ impl<S: DatabaseExt + Send, H: HistoryMode> VmTracer<EraDb<S>, H> for CheatcodeT
                     state.local_state.callstack.get_current_stack_mut().pc = pc;
                 }
                 FinishCycleOneTimeActions::ForceRevert { error, exception_handler: pc } => {
-                    tracing::debug!("!!! FORCING REVERT");
+                    tracing::debug!(?error, pc, "Forcing revert");
+
+                    self.return_data = Some(error.reason);
```

Comment:
`.to_return_data()` takes care of encoding. Probably that's why you are having padding issues and cannot exact match

Reply:
Yes! `error` is of type `ForceRevertError` (see below). In the impl there's option to instantiate using abi encoding or just the raw bytes. In the case of this PR we use `ForceRevertError::raw` given we don't want the abi encoding to return the string. But since the revert reason (same as return data) is a `Vec<U256>` it will inevitably still have padding.

```rust
#[derive(Debug, Clone)]
struct ForceRevertError {
    reason: Vec<U256>,
}

impl ForceRevertError {
    fn abi_encode(reason: Vec<u8>) -> Self {
        Self { reason: reason.to_return_data() }
    }

    fn raw(reason: Vec<u8>) -> Self {
        Self { reason: reason.chunks(32).map(U256::from_big_endian).collect_vec() }
    }
}
```
Source: https://github.com/matter-labs/foundry-zksync/blob/aon-add-assume-cheatcode/crates/era-cheatcodes/src/cheatcodes.rs#L192C1-L205C2

Reply:
But don't we actually _want_ the abi encoded reason. I can't think of anything in solidity that isn't abi encoded.

Reply:
Fixed with new commit c3cd15d